src/internal/common/network/retry_handler.go

```diff
@@ -0,0 +1,186 @@
+package network
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"syscall"
+
+	"github.com/alcionai/clues"
+)
+
+var _ io.ReadCloser = &resetRetryHandler{}
+
+const (
+	numMaxRetries  = 3
+	rangeHeaderKey = "Range"
+	// One-sided range like this is defined as starting at the given byte and
+	// extending to the end of the item.
+	rangeHeaderOneSidedValueTmpl = "bytes=%d-"
+)
+
+// Could make this per wrapper instance if we need additional flexibility
+// between callers.
+var retryErrs = []error{
+	syscall.ECONNRESET,
+}
+
+type Getter interface {
+	// SupportsRangeReq returns true if this Getter supports adding Range headers
+	// to the Get call. Otherwise returns false.
+	SupportsRangeReq() bool
+	// Get attempts to get another reader for the data this reader is returning.
+	// headers denotes any additional headers that should be added to the request,
+	// like a Range header.
+	//
+	// Don't allow passing a URL to Get so that we can hide the fact that some
+	// components may need to dynamically refresh the fetch URL (i.e. OneDrive)
+	// from this wrapper.
+	//
+	// Get should encapsulate all error handling and status code checking required
+	// for the component. This function is called both during NewResetRetryHandler
+	// and Read so it's possible to discover errors with the item prior to
+	// informing other components about it if desired.
+	Get(ctx context.Context, headers map[string]string) (io.ReadCloser, error)
+}
+
+func NewResetRetryHandler(
```

Comment:
Could you kindly add a header comment explaining that this will eagerly initialize the reader, which will in turn begin reading/downloading the Getter's data.  And that if the caller needs to defer data reads then this call must occur _within_ a lazy reader func?